src/window_func.h

```diff
@@ -56,4 +56,36 @@ void DeleteWindowByClass(WindowClass cls);
 bool EditBoxInGlobalFocus();
 Point GetCaretPosition();
 
+/**
+ * Count how many times the interval has elapsed, and update the timer.
+ * Use to ensure a specific amount of events happen within a timeframe, e.g. for animation.
+ * The timer value does not need to be initialised.
+ * @param timer Timer to test. Value will be increased.
+ * @param delta Time since last test.
+ * @param interval Timing interval.
+ * @return Number of times the interval has elapsed.
+ */
+static inline uint CountIntervalElapsed(uint &timer, uint delta, uint interval)
+{
+	uint count = delta / interval;
+	if (timer + (delta % interval) >= interval) count++;
+	timer = (timer + delta) % interval;
+	return count;
+}
+
+/**
+ * Test if a timer has elapsed, and update the timer.
+ * Use to ensure an event happens only once within a timeframe, e.g. for window updates.
+ * The timer value must be initialised in order for the timer to elapsed.
+ * @param timer Timer to test. Value will be decreased.
+ * @param delta Time since last test.
+ * @return True iff the timer has elapsed.
+ */
+static inline bool TimerElapsed(int &timer, uint delta)
```

Comment:
Not sure I like this function. From the name and the return value it just doesn't seem like it will be altering values, which would be confusing to future people. Same for `CountIntervalElapsed` above. Maybe something like `UpdateTimer` would be better?

Reply:
I'll try coming up with a better name. It needs to update the timer, as without that it's fairly pointless function.

Reply:
Although the doxygen comments do say timer will be modified.

Reply:
One option would be to use a class "GUITimer" instead of "int".
Then these tho functions would become member functions, maybe "GUITimer::CheckOnce", "GUITimer::CheckReoccuring".